### ISSUE TITLE
Is there any API documentation for this library?

### ISSUE THREAD

**Issue by jeffcharles** [wapc/as-msgpack#1]
I don't see any API documentation or example usage in the README or linked to in the description. API documentation would be helpful to use this library outside of a wapc context.

**Comment by autodidaddict** [comment]
Yeah the documentation has been "forthcoming" for a while now. @pkedy may be able to provide better information on the roadmap and timeline for the AssemblyScript MessagePack library.

**Comment by jeffcharles** [comment]
Is the library currently working? I can try and figure it out from inspecting source code but would be good to know if it's in a working state or not first.

**Comment by autodidaddict** [comment]
It's working an in use both for waPC and for the [waSCC AssemblyScript actor SDK](https://github.com/wascc/wascc-actor-as) , but the documentation for that is also mostly non-existent right now.

**Comment by chances** [comment]
@wapc Are you open to accepting PRs to add documentation? I could get something whipped up with [`typedoc`](https://typedoc.org/) pretty quickly.

**Comment by pkedy** [comment]
@chances Absolutely open to PRs!  Would love to see how `typedoc` would work here. At a minimum the README needs an example which could be derived from a waSCC/wasmCloud example or generated via the new [CLI](https://github.com/wapc/cli) using our WIDL spec (which is also in need of documentation).

**Comment by ppedziwiatr** [comment]
Can anyone please provide some example of how to use this library in a wasm module? do I need to add anything specific in the `asc` configuration to make it work? or is it just a matter of adding library in a package.json and importing it in the AS code?

**Comment by JairusSW** [comment]
I'm willing to figure out some docs and update this to v0.20.x of AS
